posts/Yoann/2023-07-07-hibernate-search-6-2-0-Final.adoc

```diff
@@ -0,0 +1,522 @@
+Specific terms that caused the match are "highlighted" with a pair of opening and closing tags.
+It can help a user to quickly identify the information they were searching for on a results page.
+
+For example, one can enable highlighting on a full-text field like this:
+
+[source, JAVA, indent=0]
+----
+@Entity
+@Indexed
+public class Book {
+    @Id
+    private Integer id;
+
+    @FullTextField(analyzer = "english")
+    private String title;
+
+    @FullTextField(analyzer = "english", projectable = Projectable.YES)
+    private String description;
+}
+----
+
+And then retrieve highlights like this:
+
+[source, JAVA, indent=0]
+----
+SearchSession searchSession = /* ... */
+
+List<List<String>> result = searchSession.search( Book.class )
+        .select( f -> f.highlight( "description" ) )
+        .where( f -> f.match().field( "description" ).matching( "robot" ) )
+        .fetchHits( 20 );
+----
+
+Which can for example return the following highlights:
+
+[source, indent=0]
+----
+# Hit #0
+["A <em>robot</em> becomes self-aware."]
+# Hit #1
+["A <em>robot</em> helps investigate a murder on an extrasolar colony.", "On this planet, <em>robots</em> are used extensively."]
+# etc.
+----
+
+This is a simple example, but highligting can be configured extensively.
+See link:{hsearch-doc-url-prefix}#search-dsl-highlighting[this section of the reference documentation]
+for more information.
+
+[[indexing-plan-filter]]
+=== Indexing plan filters
+
+Hibernate Search now allows enabling/disabling indexing in indexing plans
+(which includes in particular listener-triggered indexing),
+both fully (for all types) or on a per-type basis.
+
+For example, the following will disable automatic indexing
+for future changes on all instances of the class `Company`,
+except for instances of its subclass `Customer`:
+
+[source, JAVA, indent=0]
+----
+SearchMapping searchMapping = /* ... */
+searchMapping.indexingPlanFilter(
+        ctx -> ctx.exclude( Company.class )
+                .include( Customer.class )
+);
+----
+
+Indexing plan filters can also be configured per-session
+(though limitations apply when using the link:{hsearch-doc-url-prefix}#coordination-outbox-polling[`outbox-polling` coordination strategy]):
+
+[source, JAVA, indent=0]
+----
+SearchSession searchSession = /* ... */
+searchSession.indexingPlanFilter(
+        ctx -> ctx.exclude( Company.class )
+                .include( Customer.class )
+);
+----
+
+See link:{hsearch-doc-url-prefix}#indexing-plan-filter[this section of the reference documentation]
+for more information.
+
+[[mapping-improvements]]
+=== Mapping improvements
+
+`@IndexedEmbedded(excludePaths = ...)`::
+The `@IndexedEmbedded` annotation now exposes an `excludePaths` attribute,
+allowing the inclusion of all paths with only a few selectively excluded,
+as opposed to the previous approach of selectively including paths with `includePaths`.
```

Comment:
Is it clear that the `previous approach` is still available?

Reply:
Changed to `pre-existing`, thanks.